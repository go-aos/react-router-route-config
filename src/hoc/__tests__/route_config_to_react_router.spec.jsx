import { RouteConfig } from 'route-config'
import { oneOfType, shape, string } from 'prop-types'
import React from 'react'
import TestRenderer from 'react-test-renderer'

import Provider from '../../components/provider'
import routeConfigToReactRouter from '../route_config_to_react_router'

class Link extends React.Component {
  static displayName = 'Link'
  static propTypes = {
    to: oneOfType([shape({ pathname: string.isRequired }), string]).isRequired
  }
  render () {
    return (
      <span>
        {
          typeof this.props.to === 'string'
            ? this.props.to
            : this.props.to.pathname
        }
      </span>
    )
  }
}
const routeConfig = new RouteConfig({
  namespaces: {
    posts: {
      namespaces: {
        show: {
          path: '/:postId',
          routes: { index: {} }
        }
      },
      path: '/posts',
      routes: { index: {} }
    }
  },
  routes: {
    home: { path: '/home' }
  }
})
const RouteConfigLink = routeConfigToReactRouter(Link)
const createLink = to => TestRenderer.create(
  <Provider routeConfig={routeConfig}>
    <RouteConfigLink to={to} />
  </Provider>
)

describe('routeConfigToReactRouter hoc', () => {
  test('not change `to` props', () => {
    const link = createLink('/posts')
    expect(link.root.findByType(Link).props.to).toBe('/posts')
    expect(link.toJSON()).toMatchSnapshot()

    const objectLink = createLink({ pathname: '/posts' })
    expect(objectLink.root.findByType(Link).props.to)
      .toEqual({ pathname: '/posts' })
    expect(objectLink.toJSON()).toMatchSnapshot()
  })

  test('change `to` props', () => {
    const link = createLink('posts')
    expect(link.root.findByType(Link).props.to).toBe('/posts')
    expect(link.toJSON()).toMatchSnapshot()

    const objectLink = createLink({ pathname: 'posts' })
    expect(objectLink.root.findByType(Link).props.to)
      .toEqual({ pathname: '/posts' })
    expect(objectLink.toJSON()).toMatchSnapshot()

    const objectLinkWithParams = createLink({
      key: 'posts.show',
      params: { postId: 1 }
    })
    expect(objectLinkWithParams.root.findByType(Link).props.to)
      .toEqual({ pathname: '/posts/1' })
    expect(objectLinkWithParams.toJSON()).toMatchSnapshot()
  })
})
