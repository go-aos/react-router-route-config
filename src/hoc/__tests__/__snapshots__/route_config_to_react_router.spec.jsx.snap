// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routeConfigToReactRouter hoc change \`to\` props 1`] = `
<span>
  /posts
</span>
`;

exports[`routeConfigToReactRouter hoc change \`to\` props 2`] = `
<span>
  /posts
</span>
`;

exports[`routeConfigToReactRouter hoc change \`to\` props 3`] = `
<span>
  /posts/1
</span>
`;

exports[`routeConfigToReactRouter hoc not change \`to\` props 1`] = `
<span>
  /posts
</span>
`;

exports[`routeConfigToReactRouter hoc not change \`to\` props 2`] = `
<span>
  /posts
</span>
`;
